fix(aggregations): fix user notification aggregation

evakill opened the pull request:
PR Checklist

- [ ] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

this fixes a regression caused by https://github.com/ONEARMY/community-platform/pull/2344. We were accidentally dropping user notification aggregation updates, which prevented emails from being sent out.

i tested this fix by deploying the userNotification agg function to dev and confirming data was being populated in the `user_notifications` collection following new notifications. 

suggestions for improvement: this `functions/src/aggregations/common.aggregations.ts` file has become a little bloated with code that is not "common" to all aggregations. i think moving calculations for useful counting to the individual aggregation process functions would keep this a bit neater and easier to debug. i can draft an issue for this after lunch :) 

## Git Issues

Closes #2547 

## Screenshots/Videos

<img width="1792" alt="Screen Shot 2023-07-04 at 1 04 52 PM" src="https://github.com/ONEARMY/community-platform/assets/37253846/557fd31d-871b-433f-a8fa-0c85284f17e1">
<img width="1792" alt="Screen Shot 2023-07-04 at 2 00 01 PM" src="https://github.com/ONEARMY/community-platform/assets/37253846/89b13d5a-07fb-480b-bd8e-a1fcbf4e7856">



---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of a monthly dev call (first Monday of the month, open to all!).

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

thisislawatts commented:
Thanks @evakill, that explains the emails I got today :) Is there any testing we could introduce that would help us verify this behaviour and prevent a future regression?

evakill commented:
> Thanks @evakill, that explains the emails I got today :) Is there any testing we could introduce that would help us verify this behaviour and prevent a future regression?

thanks for the nudge to test 🙂 i found why this didn't break the existing tests: at [this fork](https://github.com/ONEARMY/community-platform/pull/2551/files#diff-b437101ae8a433e22daaf47804a022cc12e51b66308dd4db66dbfe347824173aR93) in the common.aggregation code the test DB did not have an existing `emails_pending` doc, so it called `this.seed` rather than `this.update`. the regression occurred in `this.update`. since the `emails_pending` doc already exists in all our DBs, i added it to the seed of the test so that we can test the update functionality. i also found the tests for the user notification aggregation to be a little weak so i added some cases ✨ 

i think there is definitely further testing that can/should be done in common.aggregation.spec.ts, but i don't think it should be blocking for this fix.

iSCJT reviewed (approved):
🐛 be gone
